### PR TITLE
De-duplicating SQS messages in message viewer

### PR DESCRIPTION
**Pull request by kiiadi**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Review by hunterwerlla — APPROVED** [review]

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt [hunk 1]
```diff
@@ -45,10 +41,26 @@ class PollMessagePaneTest : BaseCoroutineTest() {
...
 
-        assertThat(tableModel.items.first().messageId()).isEqualTo("XYZ")
-        assertThat(tableModel.items.first().body()).isEqualTo("ABC")
-        assertThat(tableModel.items.first().attributes().getValue(MessageSystemAttributeName.SENDER_ID)).isEqualTo("1234567890:test1")
-        assertThat(tableModel.items.first().attributes().getValue(MessageSystemAttributeName.SENT_TIMESTAMP)).isEqualTo("111111111")
+        assertThat(tableModel.items.first()).isEqualTo(message)
+    }
+
+    @Test
+    fun `Messages are de-duplicated by id`() {
```
Comment (abrooksv) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt:48]: This uses custom IDs in each message, what is causing it to test the de-dupe logic?

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt [hunk 2]
```diff
@@ -45,10 +41,26 @@ class PollMessagePaneTest : BaseCoroutineTest() {
...
+            ReceiveMessageResponse.builder().messages(message, message, anotherMessage).build()
+        )
+        val pane = PollMessagePane(projectRule.project, client, queue)
+        val tableModel = pane.messagesTable.tableModel
+        runBlocking {
+            pane.requestMessages()
+            tableModel.waitForModelToBeAtLeast(1)
+        }
+
+        assertThat(tableModel.items).hasSize(2)
```
Comment (abrooksv) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt:62]: This should be 1 if we are testing de-dupe

Comment (kiiadi) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt:62]: there's two different messages in the list `message` (x 2) and `anotherMessage`

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt (hunk 1 above)
Comment (kiiadi) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt:48]: not sure what you mean `aMessage()` creates a new message with a randomly generated id by calling `aString()`

Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt:48]: Yes, but this is called are de-duplicated by ID,

If the two messages have different IDs, this test isnt actually testing the .distincyBy

**Review by abrooksv — APPROVED** [review]

**Pull request merged**